ergo-lib/src/chain/address.rs

```diff
@@ -303,18 +368,25 @@ impl AddressEncoder {
 
     /// encode address as Base58 encoded string
     pub fn address_to_str(&self, address: &Address) -> String {
-        AddressEncoder::encode_address(self.network_prefix, &address)
+        AddressEncoder::encode_address_as_string(self.network_prefix, &address)
     }
 
     /// encode address as Base58 encoded string
-    pub fn encode_address(network_prefix: NetworkPrefix, address: &Address) -> String {
+    pub fn encode_address_as_bytes(network_prefix: NetworkPrefix, address: &Address) -> Vec<u8> {
```

Comment:
Just an FYI this is a breaking change in the API. I did this because in Yoroi we prefer handling addresses as byte arrays everywhere and this refactoring allows you to handle both bytes and base58